segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexRequestProcessor.java

```diff
@@ -51,13 +53,17 @@ public SearchFailedException(String message) {
      * @param greater boolean to determine if the next higher or the lower value to be returned in case the requested offset is not present.
      *
      * @return the corresponding offset position from the index segment entry.
-     * @throws SearchFailedException if the index segment is of unexpected size or if the search fails.
+     * @throws StreamSegmentNotExistsException If the segment is truncated or it does not exist.
      */
-    public static long locateOffsetForSegment(StreamSegmentStore store, String segment, long targetOffset, boolean greater) throws SearchFailedException {
-        return applySearch(store, segment, targetOffset, greater).getValue();
+    public static long locateOffsetForSegment(StreamSegmentStore store, String segment, long targetOffset, boolean greater) throws StreamSegmentNotExistsException {
+        try {
+            long offset = applySearch(store, segment, targetOffset, greater).getValue();
+            return offset;
+        } catch (SegmentTruncatedException ex) {
+            throw new StreamSegmentNotExistsException(ex.getMessage());
```

Comment:
These are different errors. Truncated means the segment exists but the data in it has been deleted, `StreamSegmentNotExistsException` means the segment itself does not exist. It doesn't make sense to covert between them like this.

Reply:
Reverted.